packages/dds/tree/src/simple-tree/api/typesUnsafe.ts

```diff
@@ -362,18 +364,18 @@ export type FieldHasDefaultUnsafe<T extends Unenforced<ImplicitFieldSchema>> =
  * @system @public
  */
 export type InsertableObjectFromSchemaRecordUnsafe<
-	T extends Unenforced<RestrictiveStringRecord<ImplicitFieldSchema>>,
+	T extends RestrictiveStringRecord<ImplicitFieldSchemaUnsafe>,
 > = {
 	// Field might not have a default, so make it required:
-	readonly [Property in keyof T as FieldHasDefaultUnsafe<T[Property]> extends false
+	readonly [Property in keyof T as FieldHasDefaultUnsafe<T[Property & string]> extends false
```

Comment:
Curious about this change. Do we constrain the property keys like this elsewhere? Are there other places that need this additional constraint?

Reply:
This same approach of ignoring non string keyed properties in our string keyed records is used by the safe versions of these types, and is needed here as well now that FieldHasDefaultUnsafe actually requires well typed inputs.